---
title: "HGDP1KG"
author: "Jennifer Blanc"
date: "2024-07-25"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(tidyverse)
library(maps)
library(latex2exp)
```

# Construct Test Panel

### Initial look at meta data

```{r}
# Read in data
df <- fread("../data/HGDP1KG/gnomad_meta_updated.tsv", sep = "\t")

# Select relevant columns (for now)
dfFilter <- df %>% select("s","project_meta.project_subpop" , "latitude" ,  "longitude",
                   "population",  "region", "project_meta.project_pop")
colnames(dfFilter)[1] <- "IID"

# Plot data by lat and long
world_map <- map_data("world")
pl <- ggplot() + geom_polygon(data = world_map, aes(x = long, y = lat, group = group), fill = "white", color = "black") + geom_point(data = dfFilter, aes(x = longitude, y = latitude, color = project_meta.project_pop), size = 2) +
  theme_minimal(base_size = 14) + xlab("Longitude") + ylab("Latitude") +labs(colour="POP") + theme(legend.position = "bottom")
pl
```

### Select different test panels 

**Longitude (and Latitude) in Eurasia**

Similar to the analysis in [Berg et al](https://elifesciences.org/articles/39725) we want to test for polygenic selection along longitude for populations in Eurasia (N = 2,224).   
- EAS: 825  
- FIN: 99  
- NFE: 510 (note we removed CEU samples because their lat/long corresponds to sample collection in Utah)   
- SAS: 790 

```{r}
# Select Eurasian Sample
dfEurAsia <- dfFilter %>% filter(project_meta.project_pop %in% c("fin", "nfe", "eas", "sas")) %>% 
  filter(population != "CEU") # Remove CEU due to LAT/LONG on AMERICA
dfEurAsia <- dfEurAsia %>% select("IID", "project_meta.project_pop", "population","latitude", "longitude")
colnames(dfEurAsia) <- c("IID","pop", "subpop", "lat", "long")
dfEurAsia <- dfEurAsia %>% mutate(FID = 0) %>% select(FID, everything())

# Save IDs 
fwrite(dfEurAsia,"../data/HGDP1KG/ids/test_ids/eurasia.txt", row.names = F, col.names = T, quote = F, sep = "\t")
```

```{r}
dfPlot <- dfEurAsia %>% group_by(subpop) %>% mutate(size = n()) %>% ungroup()

pl <- ggplot() + geom_polygon(data = world_map, aes(x = long, y = lat, group = group), fill = "white", color = "black") +
  geom_point(data = dfPlot, aes(x = long, y = lat, color = pop, size = size)) + 
  theme_minimal(base_size = 20) + xlab("Longitude") + ylab("Latitude") +labs(colour="Region", size = "Sample Size") + theme(legend.position = "right") + scale_color_manual(values = c( "springgreen4","slateblue4","royalblue1", "darkorange")) +  scale_size_continuous(range = c(1, 6)) 

ggsave("~/Desktop/tmp.png", height = 5, width = 12)
```



**Latitude (and Longitude) in Europe**

Similar to the analysis in [Berg et al](https://elifesciences.org/articles/39725) we want to test for polygenic selection along latitude for populations in Europe (N = 510).   
- NFE: 510 (note we removed CEU samples because their lat/long corresponds to sample collection in Utah)   

```{r}
# Select Eurasian Sample
dfEur <- dfFilter %>% filter(project_meta.project_pop %in% c("nfe"))
dfEur <- dfEur %>% select("IID", "project_meta.project_pop", "population", "latitude", "longitude") %>% 
  filter(population != "CEU")
colnames(dfEur) <- c("IID","pop", "subpop", "lat", "long")
dfEur <- dfEur %>% mutate(FID = 0) %>% select(FID, everything())

# Save IDs 
#fwrite(dfEur,"../data/HGDP1KG/ids/test_ids/nfe_no_ceu.txt", row.names = F, col.names = T, quote = F, sep = "\t")
```

```{r}
dfPlot <- dfEur %>% group_by(subpop) %>% mutate(size = n()) %>% ungroup()

pl <- ggplot() + geom_polygon(data = world_map, aes(x = long, y = lat, group = group), fill = "white", color = "black") +
  geom_point(data = dfPlot, aes(x = long, y = lat, color = pop, size = size)) + 
  theme_minimal(base_size = 20) + xlab("Longitude") + ylab("Latitude") +labs(colour="Region", size = "Sample Size") + theme(legend.position = "right") + scale_color_manual(values = c( "springgreen4","slateblue4","royalblue1", "darkorange")) +  scale_size_continuous(range = c(1, 6)) 
pl

ggsave("~/Desktop/tmp.png", height = 5, width = 12)
```



**Sardinia vs Europe**

Similar to the analysis in [Chen et al](https://www.sciencedirect.com/science/article/pii/S0002929720301610) we want to test for polygenic selection acting on height in Sardinia compared to the rest of Europe.   
- NFE: 689 (including CEU)   

```{r}
# Select Eurasian Sample
dfEur <- dfFilter %>% filter(project_meta.project_pop %in% c("nfe"))
dfEur <- dfEur %>% select("IID", "project_meta.project_pop", "population","latitude", "longitude") 
colnames(dfEur) <- c("IID", "pop", "subpop", "lat", "long")
dfEur <- dfEur %>% mutate(SDI = case_when(subpop == "Sardinian" ~ "sdi", TRUE ~ "eur")) %>% select(IID, SDI, everything())
dfEur <- dfEur %>% mutate(FID = 0) %>% select(FID, everything())

# Save IDs 
#fwrite(dfEur,"../data/HGDP1KG/ids/test_ids/nfe.txt", row.names = F, col.names = T, quote = F, sep = "\t")

```

```{r}
dfPlot <- dfEur %>% group_by(subpop) %>% mutate(size = n()) %>% ungroup()
dfPlot <- dfPlot[order(dfPlot$SDI, decreasing = FALSE), ]

pl <- ggplot() + geom_polygon(data = world_map, aes(x = long, y = lat, group = group), fill = "white", color = "black") +
  geom_point(data = dfPlot, aes(x = long, y = lat, color = SDI, size = size)) + 
  theme_minimal(base_size = 20) + xlab("Longitude") + ylab("Latitude") +labs(colour="Contrast", size = "Sample Size") + theme(legend.position = "right") + scale_color_manual(values = c( "royalblue1", "pink3")) +  scale_size_continuous(range = c(1, 6)) 
pl

ggsave("~/Desktop/tmp.png", height = 5, width = 12)
```



**All continental differences**  

Here we include the 5 major populations and will test for overlap in "continental" level contrasts (N = 3,859).  
- EAS: 825  
- NFE:  689 
- SAS: 790  
- AFR: 1003  
- AMR: 552  

```{r}
dfAll <- dfFilter %>%  filter(project_meta.project_pop %in% c("nfe", "afr", "amr", "eas", "sas")) %>% select("IID","project_meta.project_pop", "population", "latitude", "longitude")
colnames(dfAll) <- c("IID", "pop", "subpop", "lat", "long")
dfAll <- dfAll %>% mutate(FID = 0) %>% select(FID, everything())

# Save IDs 
#fwrite(dfAll,"../data/HGDP1KG/ids/test_ids/all.txt", row.names = F, col.names = T, quote = F, sep = "\t")
```

```{r}
dfPlot <- dfAll %>% group_by(subpop) %>% mutate(size = n()) %>% ungroup()

pl <- ggplot() + geom_polygon(data = world_map, aes(x = long, y = lat, group = group), fill = "white", color = "black") +
  geom_point(data = dfPlot, aes(x = long, y = lat, color = pop, size = size)) + 
  theme_minimal(base_size = 20) + xlab("Longitude") + ylab("Latitude") +labs(colour="Region", size = "Sample Size") + theme(legend.position = "right") + scale_color_manual(values = c( "purple","firebrick3","springgreen4", "royalblue1", "darkorange")) +  scale_size_continuous(range = c(1, 6)) 
pl

ggsave("~/Desktop/tmp.png", height = 5, width = 12)
```

# Results 

Load data and format 
```{r}
dfOverlap <- fread("../endpoints/HGDP1KG/OverlapStats_NewScale.txt") %>% separate(gwas, into = c("tmp", "gwas"), sep = "e")
dfOverlap$gwas <- factor(dfOverlap$gwas, levels = c( "5", "10", "50", "200")) 
dfOverlap <- dfOverlap %>% mutate(uc = Z + (1.96*(sqrt(varD))), lc = Z - (1.96*(sqrt(varD))))
```


**GWAS type vs proportion of variance explained**

Continental 
```{r}
dfCont <- dfOverlap %>% filter(contrasts %in% c("eas-nfe","eas-sas","eas-afr", "eas-amr", "nfe-sas","nfe-afr", "nfe-amr","sas-afr", "sas-amr", "afr-amr"))
dfCont$contrasts <- factor(dfCont$contrasts, levels = c("eas-nfe","eas-sas","eas-afr", "eas-amr", "nfe-sas","nfe-afr", "nfe-amr","sas-afr", "sas-amr", "afr-amr"))

pl <- dfCont %>% filter(L == "pruneall")%>% ggplot(aes(x = gwas, y = Z, color = gwas)) + geom_point(size = 2) + xlab("") + ylab("") + theme_bw(base_size = 14)  + guides(fill="none") + theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(size = 10), strip.background=element_rect(colour="black",fill="white"), legend.position = "none")+facet_wrap(~contrasts, nrow = 1)+ scale_y_log10() + geom_hline(yintercept = 1 / (99999), color = "red", linetype = "dashed") + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen")) + geom_errorbar(aes(ymin = lc, ymax = uc), width = 0.8)
pl
ggsave("~/Desktop/tmp.png", pl, height = 2.2, width = 12)
```

Lat/Long
```{r}
pl <- dfOverlap%>% filter(contrasts %in% c("eurasia-lat", "eurasia-long", "eur-lat", "eur-long", "sdi-eur")) %>% filter(L == "pruneall")%>%  ggplot(aes(x = gwas, y = Z, color = gwas)) + geom_point(size = 2) + xlab("") + ylab("") + theme_bw(base_size = 14)  + guides(fill="none") + theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(size = 10), strip.background=element_rect(colour="black",fill="white"), legend.position = "none")+facet_wrap(~contrasts, nrow = 1)+ scale_y_log10() + geom_hline(yintercept = 1 / (99999), color = "red", linetype = "dashed") + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"))
pl
ggsave("~/Desktop/tmp.png", pl, height = 2.2, width = 12)
```












```{r}
dfQ <- fread("../endpoints/HGDP1KG/QStats.txt")

pl <- dfQ %>% filter(covar == "no-FGr", contrasts %in% c("eur-lat", "eur-long"))%>% ggplot(aes(x = gwas, y = q, fill = as.character(pc))) + geom_bar(stat="identity") + facet_grid(vars(pc),vars(contrasts)) + theme_bw(base_size = 10) + theme(axis.text.x = element_text(size = 6, angle =45), plot.title = element_text(hjust = 0.5), legend.position = "none")+ scale_fill_manual(values= c("pink4", "blue4"))
pl
ggsave("~/Desktop/tmp.png",pl,  height = 4)
```


## All data together

```{r}
tmp1 <- fread("../endpoints/HGDP1KG/OverlapStats.txt")
tmp2 <- fread("../endpoints/InUKBB/OverlapStats.txt")
tmp3 <- fread("../endpoints/adnaSel/OverlapStats.txt")
tmp4 <- fread("../endpoints/SDS/OverlapStats.txt")

dfAllOverlap <- rbind(tmp1, tmp2, tmp3, tmp4) %>% separate(gwas, into = c("tmp", "gwas"), sep = "e")
dfAllOverlap$gwas <- factor(dfAllOverlap$gwas, levels = c( "5", "10", "50", "200")) 

pl <- dfAllOverlap %>% ggplot(aes(x = Z, y = error, color = gwas)) + geom_point(size = 2.5) + scale_x_log10() + scale_y_log10() + xlab("Proportion of variance explained (H)") + ylab(TeX("Error ($\\hat{e}_{F_{Gr}}$)"))  + theme_bw(base_size = 14) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)"))  + theme(panel.grid.minor = element_line(linetype = "dashed"))

#ggsave("~/Downloads/FGrError.png", pl, height = 5, width = 9)
```


```{r}
tmp1 <- fread("../endpoints/HGDP1KG/PCStats.txt")
tmp2 <- fread("../endpoints/InUKBB/PCStats.txt")
tmp3 <- fread("../endpoints/adnaSel/PCStats.txt")
tmp4 <- fread("../endpoints/SDS/PCStats.txt")

dfAllPC <- rbind(tmp1, tmp2,tmp3, tmp4) %>% separate(gwas, into = c("tmp", "gwas"), sep = "e") %>% filter(gwas_type == "even" & test_type == "odd" & L == "pruneall")
dfAllPC$gwas <- factor(dfAllPC$gwas, levels = c( "5", "10", "50", "200")) 
dfAllPC$Signal <- 1 - dfAllPC$Error
dfAllPC$Ratio <- dfAllPC$r2 / dfAllPC$Signal

# Replace ratio > 1 with 1
#dfAllPC <- dfAllPC %>% mutate(Ratio = case_when(Ratio > 1 ~ 1, Ratio <= 1 ~ Ratio))

pl <- dfAllPC %>% filter(PC == 40) %>% ggplot(aes(x = gwas, y = Ratio)) + geom_point() + facet_wrap(~contrasts)
pl
#ggsave("~/Desktop/tmp1.png", pl)
```

```{r}
dfAllC <- left_join(dfAllPC, dfAllOverlap)

pl <- dfAllC %>% filter(PC == 40) %>% ggplot(aes(x = Z, y = Ratio, color = gwas)) + geom_point() + scale_x_log10() 
pl

```


Europe Lat
```{r}
pl <- dfAllC %>% filter(contrasts == "eur-lat") %>% ggplot(aes(x = PC, y = Ratio, color = gwas)) + geom_point(size =3) + geom_line(size = 1) + facet_wrap(~contrasts, scales = "free", labeller = labeller(contrasts = c("eur-lat" = "Europe-Latitude")))  + geom_hline(yintercept = 1, color = "red", linetype = "dashed")+ scale_y_continuous(breaks = c(0,.25, 0.5, .75, 1)) + theme_bw(base_size = 20) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme(panel.grid.minor = element_line(linetype = "dashed"), legend.position = "bottom", strip.background=element_rect(colour="black",fill="white")) + ylab(TeX("$R^2_J / \\hat{\\gamma}_{F_{Gr}}$"))
pl 
ggsave("~/Desktop/tmp3.png", pl, height = 6, width = 6)
```

Everyone 
```{r}
# Continental 
dfPlot <- dfAllC %>% filter(contrasts %in%c("eas-nfe","eas-sas","eas-afr", "eas-amr", "nfe-sas","nfe-afr", "nfe-amr","sas-afr", "sas-amr", "afr-amr"))

dfPlot$contrasts <- factor(dfPlot$contrasts, levels = c("eas-nfe","eas-sas","eas-afr", "eas-amr", "nfe-sas","nfe-afr", "nfe-amr","sas-afr", "sas-amr", "afr-amr"))  

pl <- dfPlot %>% ggplot(aes(x = PC, y = Ratio, color = gwas))  + geom_point(size =1.5) + geom_line() + facet_wrap(~contrasts, labeller = labeller(contrasts = c("eur-lat" = "Europe-Latitude")), nrow =1)  + geom_hline(yintercept = 1, color = "red", linetype = "dashed")+ scale_y_continuous(breaks = c(0,.25, 0.5, .75, 1)) + theme_bw(base_size = 12) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme(panel.grid.minor = element_line(linetype = "dashed"), legend.position = "none", strip.background=element_rect(colour="black",fill="white")) + ylab(TeX("$R^2_J / \\hat{\\gamma}_{F_{Gr}}$"))

pl
ggsave("~/Desktop/tmp.png", pl, height = 2.5, width = 14)
```

```{r}
# Continental 
dfPlot <- dfAllC %>% filter(contrasts %in% c("eurasia-lat", "eurasia-long", "eur-lat", "eur-long", "sdi-eur"))


pl <- dfPlot %>% ggplot(aes(x = PC, y = Ratio, color = gwas))  + geom_point(size =1.5) + geom_line()  + facet_wrap(~contrasts, nrow =1)  + geom_hline(yintercept = 1, color = "red", linetype = "dashed")+ scale_y_continuous(breaks = c(0,.25, 0.5, .75, 1)) + theme_bw(base_size = 12) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme(panel.grid.minor = element_line(linetype = "dashed"), legend.position = "none", strip.background=element_rect(colour="black",fill="white")) + ylab(TeX("$R^2_J / \\hat{\\gamma}_{F_{Gr}}$"))
pl
ggsave("~/Desktop/tmp.png", pl, height = 2.5, width = 14)
```


```{r}
# In UK
rep_str = c("England" = "ENG","Wales" = "WAL", "Scotland" = "SCT","NorthernIreland" = "NI","RepublicOfIrelend" = "RoI"  )
dfAllC$contrasts <- str_replace_all(dfAllC$contrasts, rep_str)

dfPlot <- dfAllC %>% filter(contrasts %in% c("ENG-NI", "ENG-RoI", "ENG-SCT", "ENG-WAL", "NI-RoI", "NI-SCT", "NI-WAL", "RoI-SCT", "RoI-WAL", "SCT-WAL"))

pl <- dfPlot %>% ggplot(aes(x = PC, y = Ratio, color = gwas))  + geom_point(size =1.5) + geom_line()  + facet_wrap(~contrasts, nrow =1)  + geom_hline(yintercept = 1, color = "red", linetype = "dashed")+ scale_y_continuous(breaks = c(0,.25, 0.5, .75, 1)) + theme_bw(base_size = 12) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme(panel.grid.minor = element_line(linetype = "dashed"), legend.position = "none", strip.background=element_rect(colour="black",fill="white")) + ylab(TeX("$R^2_J / \\hat{\\gamma}_{F_{Gr}}$"))
pl
ggsave("~/Desktop/tmp.png", pl, height = 2.5, width = 14)
```


```{r}
# aDNAsel
library(ggh4x)

dfPlot <- dfAllC %>% filter(contrasts %in% c("BronzeAge", "EuropeanNeolithic", "Historical"))

custom_y <- list(
  scale_y_continuous(limits = c(0, 1)),
  scale_y_continuous(limits = c(0, 1)),
  scale_y_continuous(limits = c(0, 2.5))
)


pl <- dfPlot %>% ggplot(aes(x = PC, y = Ratio, color = gwas)) + geom_point(size =1.5) + geom_line() + facet_wrap(~contrasts, nrow =1, scales = "free_y")  + geom_hline(yintercept = 1, color = "red", linetype = "dashed") + theme_bw(base_size = 12) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme(panel.grid.minor = element_line(linetype = "dashed"), legend.position = "none", strip.background=element_rect(colour="black",fill="white")) + ylab(TeX("$R^2_J / \\hat{\\gamma}_{F_{Gr}}$")) + facetted_pos_scales(y = custom_y)
pl
ggsave("~/Desktop/tmp.png", pl, height = 2.5, width = 14)
```





```{r}
# SDS

dfPlot <- dfAllC %>% filter(contrasts %in% c("Han", "UK10K"))

custom_y <- list(
  scale_y_continuous(limits = c(0, 2.5)),
  scale_y_continuous(limits = c(0, 1))
)


pl <- dfPlot %>% ggplot(aes(x = PC, y = Ratio, color = gwas)) + geom_point(size =1.5) + geom_line() + facet_wrap(~contrasts, nrow =1, scales = "free_y")  + geom_hline(yintercept = 1, color = "red", linetype = "dashed") + theme_bw(base_size = 12) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme(panel.grid.minor = element_line(linetype = "dashed"), legend.position = "none", strip.background=element_rect(colour="black",fill="white")) + ylab(TeX("$R^2_K / \\hat{\\gamma}_{F_{Gr}}$")) + facetted_pos_scales(y = custom_y)
pl
ggsave("~/Desktop/tmp.png", pl, height = 2.5, width = 14)
```


## Q results 

```{r}
tmp1 <- fread("../endpoints/HGDP1KG/QStats.txt")
tmp2 <- fread("../endpoints/InUKBB/QStats.txt")
tmp3 <- fread("../endpoints/adnaSel/QStats.txt")
tmp4 <- fread("../endpoints/SDS/QStats.txt")

dfQ <- rbind(tmp1, tmp2, tmp3, tmp4) %>% separate(gwas, into = c("tmp", "gwas"), sep = "e") %>% filter(pc == 0 | pc == 40)
```

```{r}
tmp1 <- fread("../endpoints/HGDP1KG/PCStats.txt")
tmp2 <- fread("../endpoints/InUKBB/PCStats.txt")
tmp3 <- fread("../endpoints/adnaSel/PCStats.txt")
tmp4 <- fread("../endpoints/SDS/PCStats.txt")

dfAllPC <- rbind(tmp1, tmp2,tmp3, tmp4) %>% separate(gwas, into = c("tmp", "gwas"), sep = "e") %>% filter(gwas_type == "even" & test_type == "odd" & L == "pruneall") %>% filter(PC ==40) %>% select(!PC)
dfAllPC$gwas <- factor(dfAllPC$gwas, levels = c( "5", "10", "50", "200")) 
dfAllPC$Signal <- 1 - dfAllPC$Error
dfAllPC$Ratio <- dfAllPC$r2 / dfAllPC$Signal
```

```{r}
tmp1 <- fread("../endpoints/HGDP1KG/OverlapStats_NewScale.txt")
tmp2 <- fread("../endpoints/InUKBB/OverlapStats.txt")
tmp3 <- fread("../endpoints/adnaSel/OverlapStats.txt")
tmp4 <- fread("../endpoints/SDS/OverlapStats_NewScale.txt")

dfAllOverlap <- rbind(tmp1, tmp2, tmp3, tmp4) %>% separate(gwas, into = c("tmp", "gwas"), sep = "e")
dfAllOverlap$gwas <- factor(dfAllOverlap$gwas, levels = c( "5", "10", "50", "200")) 
```

```{r}
tmp <- left_join(dfQ, dfAllPC)
df <- left_join(tmp, dfAllOverlap)
df$HRatio <- df$Ratio * df$Z

df <- df %>% filter(Ratio < 1.5)
df$gwas<- factor(df$gwas, levels = c( "5", "10", "50", "200")) 
df$pc <-as.factor(df$pc)
```

Confounding increases with overlap but not (as much) with control 
```{r}
levels(df$gwas) <- c("5" = TeX("$5\\epsilon$"), "10" = TeX("$10\\epsilon$"), "50" = TeX("$50\\epsilon$"), "200" = TeX("$200\\epsilon$"))

pl <- df %>% ggplot(aes(x = Z, y = q^2, color = pc)) + geom_point() + geom_smooth(method = "lm", se = FALSE) + scale_x_log10()+ scale_y_log10() + facet_wrap(~gwas, scale = "free", labeller = labeller(gwas = label_parsed))  + theme_bw(base_size = 12) + ylab(TeX("$\\hat{q}^2$")) + xlab("Proportion of variance explained (H)") + scale_color_manual(values = c("goldenrod", "darkred"), name = "PCs") + theme(strip.background=element_rect(colour="black",fill="white"))
pl

ggsave("~/Desktop/allQ.png", pl, height = 6, width = 7)

#+ geom_hline(yintercept = 1, color = "blue", linetype = "dashed", size = 1.5)
```

```{r}
tmp <- df %>% filter(pc == 40 & gwas == "5 * epsilon")
lm(log10(tmp$q^2) ~ log10(tmp$Z))

lm(tmp$q^2 ~ tmp$Z)
```


HGDP1kGP
```{r}
dfCont <- df %>% filter(contrasts %in%c("eas-nfe","eas-sas","eas-afr", "eas-amr", "nfe-sas","nfe-afr", "nfe-amr","sas-afr", "sas-amr", "afr-amr", "eurasia-lat", "eurasia-long", "eur-lat", "eur-long", "sdi-eur")) %>% group_by(gwas, contrasts, pc) %>% mutate(pAdjustPheno = p.adjust(pval, method = "bonferroni")) %>% ungroup() %>% group_by(pc) %>% mutate(pAdjustAll = p.adjust(pval, method = "bonferroni"))


Sig <- dfCont %>% filter(pc == 40 & pAdjustPheno < .05)
```


Manhattan Plot
```{r}
pl <- dfCont %>% filter(pc == 40) %>% ggplot(aes(x =Ratio, y = -log10(pval), color = gwas)) + geom_point(size = 3) + geom_hline(yintercept = -log10(0.05/17), color = "gray60", linetype = "dashed", size = 1) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme_classic(base_size =20) + xlab(TeX("$R^2_{40} / \\hat{\\gamma}_{F_{Gr}}$")) + theme(legend.position = "bottom") + geom_hline(yintercept = -log10(0.05/1020), color = "lightgoldenrod3", linetype = "dashed", size = 1)
pl
#ggsave("~/Desktop/QHGDP1kGP.png", pl, width = 10, height = 5)
```

Glucose NFE-AMR
```{r}
levels(dfCont$gwas) <- c("5" = TeX("$5\\epsilon$"), "10" = TeX("$10\\epsilon$"), "50" = TeX("$50\\epsilon$"), "200" = TeX("$200\\epsilon$"))

pl <- dfCont %>% filter(contrasts == "nfe-amr" & phenotype == "Glucose") %>% mutate(upper = q + 1.96, lower = q - 1.96) %>%  ggplot(aes(x = pc, y = q, color = pc)) + geom_point(size = 3) + facet_wrap(~gwas, nrow = 1,  labeller = labeller(gwas = label_parsed)) + geom_errorbar(aes(ymin = lower, ymax = upper)) + theme_bw(base_size = 16) + scale_color_manual(values = c("goldenrod", "darkred")) + theme(legend.position = "none") + geom_hline(yintercept = 0, color = "red", linetype = "dashed") + ylab(TeX("$\\hat{q}$ (nfe - amr)")) + xlab("PCs") + theme(strip.background=element_rect(colour="black",fill="white"))
pl
ggsave("~/Desktop/glucoseQ.png", pl, width = 7, height = 2.5)
```


LDL Eurasia Long
```{r}
levels(dfCont$gwas) <- c("5" = TeX("$5\\epsilon$"), "10" = TeX("$10\\epsilon$"), "50" = TeX("$50\\epsilon$"), "200" = TeX("$200\\epsilon$"))

pl <- dfCont %>% filter(contrasts == "eurasia-long" & phenotype == "LDL") %>% mutate(upper = q + 1.96, lower = q - 1.96) %>%  ggplot(aes(x = pc, y = q, color = pc)) + geom_point(size = 3) + facet_wrap(~gwas, nrow = 1,  labeller = labeller(gwas = label_parsed)) + geom_errorbar(aes(ymin = lower, ymax = upper)) + theme_bw(base_size = 16) + scale_color_manual(values = c("goldenrod", "darkred")) + theme(legend.position = "none") + geom_hline(yintercept = 0, color = "red", linetype = "dashed") + ylab(TeX("$\\hat{q}$ (Eurasia Longitude)")) + xlab("PCs") + theme(strip.background=element_rect(colour="black",fill="white"))
pl
ggsave("~/Desktop/LDLQ.png", pl, width = 7, height = 2.5)
```


```{r}
levels(dfCont$gwas) <- c("5" = TeX("$5\\epsilon$"), "10" = TeX("$10\\epsilon$"), "50" = TeX("$50\\epsilon$"), "200" = TeX("$200\\epsilon$"))

pl <- dfCont %>% filter(contrasts == "sdi-eur" & phenotype == "StandingHeight") %>% mutate(upper = q + 1.96, lower = q - 1.96) %>%  ggplot(aes(x = pc, y = q, color = pc)) + geom_point(size = 3) + facet_wrap(~gwas, nrow = 1,  labeller = labeller(gwas = label_parsed)) + geom_errorbar(aes(ymin = lower, ymax = upper)) + theme_bw(base_size = 16) + scale_color_manual(values = c("goldenrod", "darkred")) + theme(legend.position = "none") + geom_hline(yintercept = 0, color = "red", linetype = "dashed") + ylab(TeX("$\\hat{q}$ (sdi - eur)")) + xlab("PCs")
pl
#ggsave("~/Desktop/sdiQ.png", pl, width = 7, height = 3)
```


```{r}
tmp1 <- fread("../endpoints/HGDP1KG/e5/sdi-eur/L-pruneall/StandingHeight_50/PGS/no-FGr_sdi-eur.all-PC0.sscore")
tmp1$pc <- 0
tmp2 <- fread("../endpoints/HGDP1KG/e5/sdi-eur/L-pruneall/StandingHeight_50/PGS/no-FGr_sdi-eur.all-PC40.sscore")
tmp2$pc <- 40
dfSDI <- rbind(tmp1, tmp2)
colnames(dfSDI)[1] <- "IID"
dfSDI_e5  <- left_join(dfSDI, dfEur) %>% select("IID", "pc", "SCORE1_SUM", "SDI") %>% mutate(gwas = "5")

tmp1 <- fread("../endpoints/HGDP1KG/e10/sdi-eur/L-pruneall/StandingHeight_50/PGS/no-FGr_sdi-eur.all-PC0.sscore")
tmp1$pc <- 0
tmp2 <- fread("../endpoints/HGDP1KG/e10/sdi-eur/L-pruneall/StandingHeight_50/PGS/no-FGr_sdi-eur.all-PC40.sscore")
tmp2$pc <- 40
dfSDI <- rbind(tmp1, tmp2)
colnames(dfSDI)[1] <- "IID"
dfSDI_e10  <- left_join(dfSDI, dfEur) %>% select("IID", "pc", "SCORE1_SUM", "SDI") %>% mutate(gwas = "10")

tmp1 <- fread("../endpoints/HGDP1KG/e50/sdi-eur/L-pruneall/StandingHeight_50/PGS/no-FGr_sdi-eur.all-PC0.sscore")
tmp1$pc <- 0
tmp2 <- fread("../endpoints/HGDP1KG/e50/sdi-eur/L-pruneall/StandingHeight_50/PGS/no-FGr_sdi-eur.all-PC40.sscore")
tmp2$pc <- 40
dfSDI <- rbind(tmp1, tmp2)
colnames(dfSDI)[1] <- "IID"
dfSDI_e50  <- left_join(dfSDI, dfEur) %>% select("IID", "pc", "SCORE1_SUM", "SDI") %>% mutate(gwas = "50")

tmp1 <- fread("../endpoints/HGDP1KG/e200/sdi-eur/L-pruneall/StandingHeight_50/PGS/no-FGr_sdi-eur.all-PC0.sscore")
tmp1$pc <- 0
tmp2 <- fread("../endpoints/HGDP1KG/e200/sdi-eur/L-pruneall/StandingHeight_50/PGS/no-FGr_sdi-eur.all-PC40.sscore")
tmp2$pc <- 40
dfSDI <- rbind(tmp1, tmp2)
colnames(dfSDI)[1] <- "IID"
dfSDI_e200  <- left_join(dfSDI, dfEur) %>% select("IID", "pc", "SCORE1_SUM", "SDI") %>% mutate(gwas = "200")

dfSDI <- rbind(dfSDI_e5, dfSDI_e10, dfSDI_e50, dfSDI_e200)
dfSDI$gwas<- factor(dfSDI$gwas, levels = c( "5", "10", "50", "200")) 
#levels(dfSDI$gwas) <- c("5" = TeX("$5\\epsilon$"), "10" = TeX("$10\\epsilon$"), "50" = TeX("$50\\epsilon$"), "200" = TeX("$200\\epsilon$"))
```

```{r}
levels(dfSDI$gwas) <- c("5" = TeX("$5\\epsilon$"), "10" = TeX("$10\\epsilon$"), "50" = TeX("$50\\epsilon$"), "200" = TeX("$200\\epsilon$"))

pl <- dfSDI %>%
  ggplot(aes(x = SDI, y = SCORE1_SUM, group = SDI, fill = SDI)) + 
  geom_violin(aes(fill = SDI), alpha = 0.2) +  # Makes the violin plot transparent
  facet_grid(vars(pc), vars(gwas), scales = "free",  labeller = labeller(gwas = label_parsed, pc = c("0" = "0 PCs", "40" = "40 PCs"))) + stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  theme_bw(base_size = 16) + ylab("Height polygenic score") + xlab("") + theme(legend.position = "none") +
  scale_fill_manual(values = c("royalblue1", "pink3"))

ggsave("~/Desktop/sdiPGS.png", pl, width = 7, height = 5)
```


Within UK
```{r}
rep_str = c("England" = "ENG","Wales" = "WAL", "Scotland" = "SCT","NorthernIreland" = "NI","RepublicOfIrelend" = "RoI"  )
df$contrasts <- str_replace_all(df$contrasts, rep_str)

dfUK <- df %>% filter(contrasts %in% c("ENG-NI", "ENG-RoI", "ENG-SCT", "ENG-WAL", "NI-RoI", "NI-SCT", "NI-WAL", "RoI-SCT", "RoI-WAL", "SCT-WAL")) %>% group_by(gwas, contrasts, pc) %>% mutate(pAdjust = p.adjust(pval, method = "bonferroni"))

Sig <- dfUK %>% filter(pc == 40 & pAdjust < .05)
```

```{r}
pl <- dfUK %>% filter(pc == 40) %>% ggplot(aes(x =Ratio, y = -log10(pval), color = gwas)) + geom_point(size = 3) + geom_hline(yintercept = -log10(0.05/17), color = "gray60", linetype = "dashed", size = 1) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme_classic(base_size =20) + xlab(TeX("$R^2_K / \\hat{\\gamma}_{F_{Gr}}$")) + theme(legend.position = "bottom")
pl

#ggsave("~/Desktop/QHGDP1kGP.png", pl, width = 10, height = 5)
```

```{r}
df %>% filter(contrasts == "ENG-SCT" & phenotype == "MeanCorpuscularVolume") %>% ggplot(aes(x = gwas, y = q, color = as.character(pc))) + geom_point() 
```

aDNA
```{r}

dfaDNA <- df %>% filter(contrasts %in% c("Historical", "BronzeAge", "EuropeanNeolithic")) %>% group_by(gwas, contrasts, pc) %>% mutate(pAdjust = p.adjust(pval, method = "bonferroni"))

Sig <- dfaDNA %>% filter(pc == 40 & pAdjust < .05)
```

```{r}
pl <- dfaDNA %>% filter(pc == 40) %>% ggplot(aes(x =Ratio, y = -log10(pval), color = gwas, shape = contrasts)) + geom_point(size = 3) + geom_hline(yintercept = -log10(0.05/17), color = "gray60", linetype = "dashed", size = 1) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme_classic(base_size =20) + xlab(TeX("$R^2_K / \\hat{\\gamma}_{F_{Gr}}$")) + theme(legend.position = "bottom") + guides(color = "none")
pl

#ggsave("~/Desktop/QHGDP1kGP.png", pl, width = 10, height = 5)
```


SDS
```{r}

dfSDS <- df %>% filter(contrasts %in% c("UK10K", "Han")) %>% group_by(gwas, contrasts, pc) %>% mutate(pAdjust = p.adjust(pval, method = "bonferroni"))

Sig <- dfSDS %>% filter(pc == 40 & pAdjust < .05)
```

```{r}
pl <- dfSDS %>% filter(pc == 40) %>% ggplot(aes(x =Ratio, y = -log10(pval), color = gwas)) + geom_point(size = 3) + geom_hline(yintercept = -log10(0.05/17), color = "gray60", linetype = "dashed", size = 1) + scale_color_manual(values = c("purple1", "darkorange", "skyblue2", "seagreen"), name= TeX("Sampling distance ($\\epsilon$)")) + theme_classic(base_size =20) + xlab(TeX("$R^2_K / \\hat{\\gamma}_{F_{Gr}}$")) + theme(legend.position = "bottom")
pl

#ggsave("~/Desktop/QHGDP1kGP.png", pl, width = 10, height = 5)
```






```{r}
L <- 500
M <- 100


# Make Genotype Matrix
G <- matrix(NA, nrow = M, ncol = L)
for (i in 1:L) {
  G[,i] <- c(rnorm(50, mean = 0, sd = 1), rnorm(50, mean = 0, sd = 1))
}
G <- scale(G)
 

# Make r 
r <- c(rnorm(250, mean = 0, sd = 1), rnorm(250, mean = 0, sd = 1))
r <- r - mean(r)
r <- r / sqrt(sum(r^2))

#plot(G[,1])

# Calculate R
R <- (1/(M-1)) * (t(G) %*% G)
sum(diag(R))
myR <- eigen(R)
sum(myR$values)

# Calculate K
K <- (t(G) %*% G)
sum(diag(K))
myK <- eigen(K)
sum(myK$values)

# Calculate D 
D <- t(r) %*% t(G) %*% G %*% r 

D
```



```{r}
tmp <- rep(0, M)
for (i in 1:M){
  tmp[i] <- cov(G[i,],r)^2
}
mean(tmp)
(1/M) * (1/(L-1))^2 * (D)


tmp <- rep(0, M)
for (i in 1:M){
  tmp[i] <- var(G[i,])
}
mean(tmp)
sum(diag((1/(L-1)) *G %*% t(G))) * (1/M)

((1/(L-1))^2 * (D)) / (sum(diag((1/(L-1)) *G %*% t(G))))
```

```{r}
simulate_block <- function(size) {
  # Create a random matrix
  A <- matrix(rnorm(size^2), nrow = size)
  # Symmetrize the matrix to ensure it's a valid covariance matrix
  cov_block <- A %*% t(A)
  # Optionally, scale the covariance matrix to control the variances
  cov_block <- cov_block / max(diag(cov_block))
  return(cov_block)
}

b1 <- matrix(rep(1, 50^2),50,50)
b2 <- matrix(rep(0.5, 50^2),50,50)
Sigma1 <- rbind(b1, b2)
Sigma2 <- rbind(b2, b1)
Sigma <- cbind(Sigma1, Sigma2)
Sigma <- matrix(rep(1, 100^2),100,100)

G <- matrix(NA, nrow = M, ncol = L)
for (i in 1:L) {
  G[,i ] <- t(mvrnorm(n = 1, rep(0, M), Sigma = Sigma))
}
G <- t(mvrnorm(n = 1, rep(0, M), Sigma = Sigma))
X <-  t(rmvnorm(n = L, rep(0, M), Sigma = Sigma))

r <- c(rep(-1, M/2), rep(1, M/2)) %*% X
r <- r[1,]
r <- r - mean(r)
r <- r / sqrt(sum(r^2))

G <- scale(G)
FGr <- G %*% r

H <- t(FGr) %*% FGr * (1/(M-1)) * (1/L)
expH <- 1/(M-1)

# Create a heatmap without row/column reordering
heatmap(G %*% t(G), Rowv = NA, Colv = NA, scale = "none")

Sigma <- matrix(c(10,3,3,2),2,2)
Sigma
mvrnorm(n = 1000, rep(0, 2), Sigma)

```



```{r}
Tm_prefix <- "~/Downloads/FGr_eas-nfe"
snp_prefix <- "~/Downloads/eas-nfe"

# Get list of all chromosomes
chrs <- seq(1,22)

# Add Tm for each chromosome to each other
data <- fread(paste0(Tm_prefix, "_1.txt"))
data <- data[,4:ncol(data)]

for (i in 2:22) {

  print(paste0("chr num ",i))
  # Read in new chromosome
  filename <- paste0(Tm_prefix,"_", i, ".txt")
  tmp <- fread(filename)
  tmp <- tmp[,4:ncol(tmp)]
  data <- cbind(data, tmp)

}
data <- as.data.frame(data)
print(dim(data))

# Find total number of SNPs
snp_nums <- fread(paste0(snp_prefix, "_1_SNPcount.txt"))
for (i in 2:22) {

  print(paste0("chr num ",i))
  # Read in new chromosome
  snp_nums <- rbind(snp_nums, fread(paste0(snp_prefix,"_", i, "_SNPcount.txt")))

}
print(dim(snp_nums))
print(snp_nums)

# Set parameters
L <- sum(snp_nums$nSNP)
M <- nrow(data)
print(paste0("L is ", L))
print(paste0("M is ", M))

```


```{r}
FGr_hat <- apply(data, 1, sum) * (1/sqrt(L))
var(FGr_hat)
D <- t(FGr_hat) %*% FGr_hat *(1 / M)

* (1/L) 

* (1/L)

D
```



