---
title: "SDS"
author: "Jennifer Blanc"
date: "2024-07-19"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(tidyverse)
library(ggpubr)
library(latex2exp)
```

# Construct Test and GWAS panels 

### Explore SDS data

Read in both files
```{r, warning=FALSE, message=FALSE}
UK <- fread("../data/SDS/raw/SDS_UK10K_n3195_release_Sep_19_2016.tab.gz")
Han <- fread("../data/SDS/raw/auto_SDS_norm_sort.txt.gz")
```

Look at formatting
```{r}
head(UK)
```

```{r}
head(Han)
```

Convert Han coordinates to hg19
```{r, eval=FALSE}
library(rtracklayer)
library(GenomicRanges)

#chain_url <- "http://hgdownload.cse.ucsc.edu/goldenPath/hg38/liftOver/hg38ToHg19.over.chain.gz"
#chain_file <- "hg38ToHg19.over.chain.gz"
#download.file(chain_url, chain_file)

# Download the chain file
chain_file <- "../data/liftover/hg38ToHg19.over.chain.gz"
#download.file(chain_url, chain_file, mode = "wb")

# Uncompress the chain file
chain_file_unzipped <- sub(".gz$", "", chain_file)
#R.utils::gunzip(chain_file, chain_file_unzipped, remove = FALSE)

# Load the chain file
chain <- import.chain(chain_file_unzipped)

# Create a GRanges object with your SNP coordinates
Han$chr <- paste0("chr", Han$chr)
gr <- GRanges(seqnames = Han$chr, ranges = IRanges(start = Han$pos, end = Han$pos))

# Perform the liftOver
lifted_snps <- liftOver(gr, chain)

# Extract the results
converted_snps <- as.data.frame(unlist(lifted_snps))
converted_snps

# Filter out empty liftOver results
non_empty_indices <- elementNROWS(lifted_snps) > 0
filtered_gr <- gr[non_empty_indices]
filtered_lifted_snps <- lifted_snps[non_empty_indices]

# Convert the filtered GRanges and lifted results to data frames
original_coords <- as.data.frame(filtered_gr)
lifted_coords <- as.data.frame(unlist(filtered_lifted_snps))

# Combine the data frames
combined_coords <- data.frame(
  original_chr = original_coords$seqnames,
  original_start = original_coords$start,
  original_end = original_coords$end,
  lifted_chr = lifted_coords$seqnames,
  lifted_start = lifted_coords$start,
  lifted_end = lifted_coords$end
)

```

Join lifted over coordinates with SDS values and save
```{r, eval=FALSE}
dfHan <- inner_join(Han, combined_coords, by = c("chr" = "original_chr", "pos" = "original_start"))

#fwrite(dfHan,"../data/SDS/raw/Han_SDS_liftedover.txt.gz", row.names = F, col.names = T, quote = F, sep = "\t")
```


# Results

### Overlap statistics


Load data and format 
```{r}
dfOverlap <- fread("../endpoints/SDS/OverlapStats.txt") %>% separate(gwas, into = c("tmp", "gwas"), sep = "e")
dfOverlap$gwas <- factor(dfOverlap$gwas, levels = c( "5", "10", "50", "200"))
```

Plot Z by GWAS
```{r}
pl <- dfOverlap %>% filter(L == "pruneall")%>% ggplot(aes(x = gwas, y = Z)) + geom_point() + xlab("") + ylab("") + theme_bw(base_size = 14)  + guides(fill="none") + theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(size = 10), strip.text.x = element_text(size = 8,))+facet_wrap(~contrasts, nrow = 1)+ scale_y_log10() 
pl
ggsave("~/Desktop/tmp.png", pl, height = 2, width = 12)
```

Load PC data 
```{r}
dfPC <- fread("../endpoints/SDS/PCStats.txt") %>% separate(gwas, into = c("tmp", "gwas"), sep = "e")
dfPC$gwas <- factor(dfPC$gwas, levels = c( "5", "10", "50", "200"))
```

```{r}
dfPC %>% filter(gwas_type == "even" & test_type == "odd" & L == "pruneall") %>% ggplot(aes(x = PC, y = r2)) + geom_bar(stat = "identity", fill = "darkgreen", color = "darkgreen") + facet_grid(vars(gwas), vars(contrasts)) + theme_bw(base_size = 8) + ylab(TeX("$R_{K}^2$")) + geom_hline( aes(yintercept = 1-Error), color = "red", linetype = 2, linewidth = 1) + scale_x_continuous(breaks = seq(0, 40, by = 10)) + theme(axis.text.x = element_text(size = 6), plot.title = element_text(hjust = 0.5)) + ylim(-0.1,1)
```

```{r}
dfGap <-  dfPC %>% filter(gwas_type == "even" & test_type == "odd" & L == "pruneall" & PC == 40) %>% mutate(signal = 1- Error) %>% mutate(gap40 = r2/ signal) %>% select(gwas, contrasts, gap40)

dfQ40 <- dfQ %>% filter(pc == 40) %>% select(q, gwas, contrasts, phenotype) 
dfQ0 <- dfQ %>% filter(pc == 0) %>% select(q, gwas, contrasts, phenotype)

dfPlot <- left_join(dfQ0, dfGap)
dfPlot$predQ <- dfPlot$q * dfPlot$gap40

dfPlot2 <- inner_join(dfPlot, dfQ40, by = c("gwas", "contrasts", "phenotype"))

dfPlot2 %>% filter(contrasts == "UK10K") %>% ggplot(aes(x = predQ , y = q.y )) + geom_point()
```




```{r}
dfQ <- fread("../endpoints/SDS/QStats.txt") %>% separate(gwas, into = c("tmp", "gwas"), sep = "e")
dfQ$gwas <- factor(dfQ$gwas, levels = c( "5", "10", "50", "200"))

dfQ %>% filter(covar == "no-FGr" & phenotype == "BasophillPercentage" )%>% ggplot(aes(x = gwas, y = q, fill = as.character(pc))) + geom_bar(stat="identity") + facet_grid(vars(pc),vars(contrasts)) + theme_bw(base_size = 10) + theme(axis.text.x = element_text(size = 6, angle =45), plot.title = element_text(hjust = 0.5), legend.position = "none")+ scale_fill_manual(values= c("pink4", "blue4", "purple"))
```

```{r}
dfPlot <- left_join(dfQ, dfOverlap)

dfPlot %>% ggplot(aes(x = Z, y = abs(q), color = contrasts)) + geom_point() + facet_wrap(~pc)
```

```{r}
dfTmp <- dfQ %>% group_by(contrasts, gwas, pc) %>% mutate(pAdjust = p.adjust(pval, method = "bonferroni"))

dfSigUK <- dfTmp %>% filter(contrasts == "UK10K" & pAdjust < 0.05 & pc == 40) %>% select(q, pval, nsnp, gwas, pAdjust, phenotype)
dfSigUK

## No significant results for Han 
dfSigHan <- dfTmp %>% filter(contrasts == "Han" & pAdjust < 0.05 & pc == 40) %>% select(q, pval, nsnp, gwas, pAdjust, phenotype)
#dfSigHan
```

